### PR TITLE
Remove res_config getters from EnKFMain

### PR DESCRIPTION
**Pull request by oyvindeide**
Delete some unused code and simplify.

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

**Review by eivindjahren — APPROVED** [review]
Excellent job!

**Comment by sondreso** [comment]
Could you update the PR title for the changelog? :sweat_smile:

File: src/res/enkf/enkf_main.py [hunk 1]
```diff
@@ -396,26 +373,26 @@ def getEnsembleSize(self) -> int:
 
     def ensembleConfig(self) -> EnsembleConfig:
         """@rtype: EnsembleConfig"""
```
Comment (eivindjahren) [inline src/res/enkf/enkf_main.py:375]: Maybe just fix up the rtype docstrings while we are here. They are superseded by type annotations.

File: src/res/enkf/enkf_main.py [hunk 2]
```diff
@@ -443,11 +420,8 @@ def have_observations(self):
     def loadObservations(self, obs_config_file, clear=True):
         return self._load_obs(obs_config_file, clear)
 
-    def get_site_config_file(self):
-        return self._get_site_config_file()
-
     def getHistoryLength(self):
```
Comment (eivindjahren) [inline src/res/enkf/enkf_main.py:423]: Could add a type annotation here.

File: src/libres/lib/include/ert/enkf/enkf_main.hpp [hunk 3]
```diff
@@ -118,21 +103,15 @@ enkf_main_load_from_run_context(enkf_main_type *enkf_main,
                                     std::vector<bool> active_mask,
                                     enkf_fs_type *sim_fs,
                                     std::vector<run_arg_type*> run_args);
-
+char *enkf_main_read_alloc_current_case_name(const enkf_main_type *enkf_main);
```
Comment (eivindjahren) [inline src/libres/lib/include/ert/enkf/enkf_main.hpp:106]: Seems like this is only changing location in the diff.

**Review by eivindjahren — CHANGES_REQUESTED** [review]
Only minor comments :)

File: src/libres/lib/include/ert/enkf/enkf_main.hpp (hunk 3 above)
Comment (oyvindeide) [reply src/libres/lib/include/ert/enkf/enkf_main.hpp]: Yes, moved it back :+1:

**Review by eivindjahren — APPROVED** [review]
:rocket:

**Pull request merged**